bundles/org.openhab.binding.e3dc/src/main/java/org/openhab/binding/e3dc/internal/handler/E3DCDeviceThingHandler.java

```diff
@@ -0,0 +1,178 @@
+                        REGISTER_LENGTH - INFO_REG_SIZE, 3);
+                dataPoller = localModbusCom.registerRegularPoll(dataRequest, localConfig.refresh, 0, modbusDataCallback,
+                        modbusDataCallback);
+                modbusCom = localModbusCom;
+                // listen for data to get ONLINE
+                modbusDataCallback.addDataListener(this);
+            } else {
+                setStatus(ThingStatus.OFFLINE);
+            }
+        });
+    }
+
+    @Override
+    public void dispose() {
+        modbusDataCallback.removeDataListener(this);
+        ModbusCommunicationInterface localCom = modbusCom;
+        if (localCom != null) {
+            PollTask localInfoPoller = infoPoller;
+            if (localInfoPoller != null) {
+                localCom.unregisterRegularPoll(localInfoPoller);
+            }
+            PollTask localDataPoller = dataPoller;
+            if (localDataPoller != null) {
+                localCom.unregisterRegularPoll(localDataPoller);
+            }
+        }
+    }
+
+    private boolean checkConfig(@Nullable E3DCDeviceConfiguration c) {
+        if (c != null) {
+            if (c.port > 1) {
+                if (c.refresh < 1) {
+                    c.refresh = 2;
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void setStatus(ThingStatus status) {
+        myStatus = status;
+        updateStatus(myStatus);
+    }
+
+    /**
+     * Wallbox Settings can be changed with one Integer
+     *
+     * @param wallboxId needed to calculate right register
+     * @param writeValue integer to be written
+     */
+    public void wallboxSet(int wallboxId, int writeValue) {
+        E3DCDeviceConfiguration localConfig = config;
+        ModbusCommunicationInterface localCom = modbusCom;
+        if (localConfig != null && localCom != null) {
+            ModbusRegisterArray regArray = new ModbusRegisterArray(writeValue);
+            ModbusWriteRegisterRequestBlueprint writeBluePrint = new ModbusWriteRegisterRequestBlueprint(
+                    localConfig.deviceid, WALLBOX_REG_START + wallboxId, regArray, false, 3);
+            localCom.submitOneTimeWrite(writeBluePrint, this, this);
+        }
+    }
+
+    @Override
+    public void dataAvailable(ModbusDataProvider provider) {
+        if (myStatus != ThingStatus.ONLINE) {
+            setStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    public ModbusDataProvider getInfoDataProvider() {
+        return modbusInfoCallback;
+    }
+
+    public ModbusDataProvider getDataProvider() {
+        return modbusDataCallback;
+    }
+
+    @Override
+    public void handle(AsyncModbusWriteResult result) {
+        logger.debug("E3DC Modbus write response! {}", result.getResponse().toString());
+    }
+
+    @Override
+    public void handle(AsyncModbusFailure<ModbusWriteRequestBlueprint> failure) {
```

Comment:
error essentially ignored -- would you not want to change thing status?

Reply:
Error handling is completely reworked